### ISSUE TITLE
retraining with custom dataset EfficientNet-v2

### ISSUE THREAD

**Issue by keertika-11** [google/automl#1013]
I want to load the pre-trained weights and want to finetune on custom dataset.

**Comment by akashAD98** [comment]
I'm also looking for the same.if anyone trained their own custom data with efficient netv2. please attach reference here

**Comment by panomwat-w** [comment]
I cannot provide reference but a suggestion that for a custom dataset you should modify something
1. create your own custom dataset class in datasets.py and you may do so by copy a class ImageNetInput() and modify
2. for my project, I've created TFRecord files and pass it in the dataset class by data_dir (you may want to check this out to create TFRecord file https://github.com/tensorflow/tpu/blob/master/tools/datasets/imagenet_to_gcs.py)

**Comment by mingxingtan** [comment]
you can see the colab for an example: https://colab.sandbox.google.com/github/google/automl/blob/master/efficientnetv2/tutorial.ipynb

**Comment by keertika-11** [comment]
the tutorial refers to pre-defined datasets, how can it be done for custom datasets?

**Comment by lunzueta** [comment]
> 
> 
> I cannot provide reference but a suggestion that for a custom dataset you should modify something
> 
>     1. create your own custom dataset class in datasets.py and you may do so by copy a class ImageNetInput() and modify
> 
>     2. for my project, I've created TFRecord files and pass it in the dataset class by data_dir (you may want to check this out to create TFRecord file https://github.com/tensorflow/tpu/blob/master/tools/datasets/imagenet_to_gcs.py)

I've tried this, but I get an error when loading the dataset with tfds.load(). It says that it's not available and it gives me a list of those that are available online. How can custom data stored in your disk be loaded?

**Comment by keertika-11** [comment]
import tensorflow as tf
import effnetv2_model

input_layer = tf.keras.layers.Input(shape=(224, 224, 3))
model = effnetv2_model.get_model('efficientnetv2-b0', include_top=False,weights='imagenet')(input_layer)
dropout = tf.keras.layers.Dropout(rate=0.2)(model)
out = tf.keras.layers.Dense(1,activation='sigmoid')(dropout)
final_model = tf.keras.models.Model(inputs=input_layer,outputs=out)
final_model.summary()

**Comment by keertika-11** [comment]
```
import tensorflow as tf
import os
import random
from glob import glob


def make_dataset(path, batch_size):
    def preprocess_image(filename):
        image = tf.io.read_file(filename)
        image = tf.image.decode_jpeg(image, channels=3)
        image = tf.image.convert_image_dtype(image, tf.float32)
        image = image / 255.0
        image = tf.image.resize(image, [224, 224])
        return image

    def configure_for_performance(dataset):
        """
        examples.prefetch(2) will prefetch two elements (2 examples),
        while examples.batch(20).prefetch(2) will prefetch 2 elements
        (2 batches, of 20 examples each)

        train_ds = train_ds.cache().shuffle(1000).prefetch(buffer_size=AUTOTUNE)
        """

        dataset = dataset.cache('/tmp/dump.tfcache')
        dataset = dataset.shuffle(buffer_size=1000)
        dataset = dataset.batch(batch_size)
        dataset = dataset.repeat()
        dataset = dataset.prefetch(buffer_size=tf.data.experimental.AUTOTUNE)

        return dataset

    def image_preprocess(image):
        image = tf.image.random_flip_left_right(image)
        image = tf.image.random_brightness(image, max_delta=32.0 / 255.0)
        image = tf.image.random_saturation(image, lower=0.5, upper=1.5)
        image = tf.clip_by_value(image, 0.0, 1.0)
        return image

    classes = os.listdir(path)
    filenames = glob(path + '/*/*')
    random.shuffle(filenames)
    labels = [classes.index(name.split('/')[-2]) for name in filenames]

    image_data = tf.data.Dataset.from_tensor_slices(filenames)
    preprocessed_image_data = image_data.map(preprocess_image, num_parallel_calls=tf.data.experimental.AUTOTUNE)
    labels = tf.data.Dataset.from_tensor_slices(labels)
    image_label_dataset = tf.data.Dataset.zip((preprocessed_image_data, labels))
    image_label_dataset = configure_for_performance(image_label_dataset)

    return image_label_dataset


if __name__ == '__main__':
    import os
    train_dataset = make_dataset("sample/train", 32)
    test_dataset = make_dataset("sample/test", 1)

```

> > I cannot provide reference but a suggestion that for a custom dataset you should modify something
> > ```
> > 1. create your own custom dataset class in datasets.py and you may do so by copy a class ImageNetInput() and modify
> > 
> > 2. for my project, I've created TFRecord files and pass it in the dataset class by data_dir (you may want to check this out to create TFRecord file https://github.com/tensorflow/tpu/blob/master/tools/datasets/imagenet_to_gcs.py)
> > ```
> 
> I've tried this, but I get an error when loading the dataset with tfds.load(). It says that it's not available and it gives me a list of those that are available online. How can custom data stored in your disk be loaded?

**Comment by lunzueta** [comment]
> 
> 
> > I cannot provide reference but a suggestion that for a custom dataset you should modify something
> > ```
> > 1. create your own custom dataset class in datasets.py and you may do so by copy a class ImageNetInput() and modify
> > 
> > 2. for my project, I've created TFRecord files and pass it in the dataset class by data_dir (you may want to check this out to create TFRecord file https://github.com/tensorflow/tpu/blob/master/tools/datasets/imagenet_to_gcs.py)
> > ```
> 
> I've tried this, but I get an error when loading the dataset with tfds.load(). It says that it's not available and it gives me a list of those that are available online. How can custom data stored in your disk be loaded?

In the end, I was able to insert my custom dataset following [this procedure](https://www.tensorflow.org/datasets/add_dataset), and then editing the datasets.py file accordingly to recover it within automl. No need to convert to .tfrecord by myself, this is already done when doing `tfds build`.
